server/src/main/java/com/datastrato/gravitino/server/web/rest/UserOperations.java

```diff
@@ -0,0 +1,123 @@
+import javax.ws.rs.Produces;
+import javax.ws.rs.core.Context;
+import javax.ws.rs.core.Response;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+@Path("/metalakes/{metalake}/users")
+public class UserOperations {
+
+  private static final Logger LOG = LoggerFactory.getLogger(UserOperations.class);
+
+  private final AccessControlManager accessControlManager;
+
+  @Context private HttpServletRequest httpRequest;
+
+  @Inject
+  public UserOperations(AccessControlManager accessControlManager) {
+    this.accessControlManager = accessControlManager;
+  }
+
+  @GET
+  @Path("{user}")
+  @Produces("application/vnd.gravitino.v1+json")
+  @Timed(name = "get-user." + MetricNames.HTTP_PROCESS_DURATION, absolute = true)
+  @ResponseMetered(name = "get-user", absolute = true)
+  public Response getUser(@PathParam("metalake") String metalake, @PathParam("user") String user) {
+    try {
+      NameIdentifier ident = ofUser(metalake, user);
+      return Utils.doAs(
+          httpRequest,
+          () ->
+              Utils.ok(
+                  new UserResponse(
+                      DTOConverters.toDTO(
+                          TreeLockUtils.doWithTreeLock(
+                              ident,
+                              LockType.READ,
+                              () -> accessControlManager.getUser(metalake, user))))));
+    } catch (Exception e) {
+      return ExceptionHandlers.handleUserException(OperationType.GET, user, metalake, e);
+    }
+  }
+
+  @POST
+  @Produces("application/vnd.gravitino.v1+json")
+  @Timed(name = "add-user." + MetricNames.HTTP_PROCESS_DURATION, absolute = true)
+  @ResponseMetered(name = "add-user", absolute = true)
+  public Response addUser(@PathParam("metalake") String metalake, UserAddRequest request) {
+    try {
+      NameIdentifier ident = ofUser(metalake, request.getName());
+      return Utils.doAs(
+          httpRequest,
+          () ->
+              Utils.ok(
+                  new UserResponse(
+                      DTOConverters.toDTO(
+                          TreeLockUtils.doWithTreeLock(
+                              ident,
+                              LockType.WRITE,
+                              () -> accessControlManager.addUser(metalake, request.getName()))))));
+    } catch (Exception e) {
+      return ExceptionHandlers.handleUserException(
+          OperationType.ADD, request.getName(), metalake, e);
+    }
+  }
+
+  @DELETE
+  @Path("{user}")
+  @Produces("application/vnd.gravitino.v1+json")
+  @Timed(name = "remove-user." + MetricNames.HTTP_PROCESS_DURATION, absolute = true)
+  @ResponseMetered(name = "remove-user", absolute = true)
+  public Response removeUser(
+      @PathParam("metalake") String metalake, @PathParam("user") String user) {
+    try {
+      return Utils.doAs(
+          httpRequest,
+          () -> {
+            NameIdentifier ident = ofUser(metalake, user);
+            boolean removed =
+                TreeLockUtils.doWithTreeLock(
```

Comment:
@yuqi1129 Could you help me review the use of lock?

Reply:
This part maybe changed after #2732 , let's wait for a while.